core/templates/dev/head/exploration_editor/EditorServices.js

```diff
@@ -1839,3 +1890,354 @@ oppia.factory('explorationWarningsService', [
+        break;
+      case 'widget_id':
+        if (oldValue === null) {
+          if (newValue !== 'EndExploration') {
+            label = 'Added Interaction ' + newValue;
+            type = TYPE_ADD_INTERACTION;
+          } else {
+            label = 'Ended Exploration';
+            type = TYPE_END_EXPLORATION;
+          }
+        } else {
+          label = 'Deleted Interaction ' + oldValue;
+          type = TYPE_DELETE_INTERACTION;
+        }
+        break;
+      case 'widget_customization_args':
+        if (_isEmpty(oldValue)) {
+          label = 'Added Interaction Customizations';
+          type = TYPE_ADD_WIDGET_CUSTOMIZATIONS;
+        } else if (_isEmpty(newValue)) {
+          label = 'Removed Interaction Customizations';
+          type = TYPE_DELETE_WIDGET_CUSTOMIZATIONS;
+        } else {
+          label = 'Edited Interaction Customizations';
+          type = TYPE_EDIT_WIDGET_CUSTOMIZATIONS;
+        }
+        break;
+      case 'answer_groups':
+        var answerGroupChanges = _getRelativeChangeToGroups(lostChange);
+        if (answerGroupChanges === 'added') {
+          label = 'Added answer group';
+          type = TYPE_ADD_ANSWER_GROUP;
+          value = _getAnswerGroupOrDefaultOutcomeDetails(
+            lostChange, 'answer_group', newValue);
+        } else if (answerGroupChanges === 'edited') {
+          label = 'Edited answer group.';
+          type = TYPE_EDIT_ANSWER_GROUP;
+          value = _getAnswerGroupOrDefaultOutcomeEdits(
+            lostChange, 'answer_group', newValue, oldValue);
+        } else {
+          label = 'Deleted answer group';
+          type = TYPE_DELETE_ANSWER_GROUP;
+        }
+        break;
+      case 'default_outcome':
+        var defaultOutcomeChanges = _getRelativeChangeToGroups(lostChange);
+        if (defaultOutcomeChanges === 'added') {
+          label = 'Added default outcome';
+          type = TYPE_ADD_DEFAULT_OUTCOME;
+          value = _getAnswerGroupOrDefaultOutcomeDetails(
+            lostChange, 'default_outcome', newValue);
+        } else if (defaultOutcomeChanges === 'edited') {
+          label = 'Edited default outcome';
+          type = TYPE_EDIT_DEFAULT_OUTCOME;
+          value = _getAnswerGroupOrDefaultOutcomeEdits(
+            lostChange, 'default_outcome', newValue, oldValue);
+        } else {
+          label = 'Deleted default outcome';
+          type = TYPE_DELETE_DEFAULT_OUTCOME;
+        }
+        break;
+    };
+
+    return _createChangeObject(label, type, value);
+  };
+
+  var _getStatePropertyValue = function(
+    statePropertyValue) {
+    return angular.isArray(statePropertyValue) ?
+      statePropertyValue[statePropertyValue.length - 1] : statePropertyValue;
+  };
+
+  var _isEmpty = function(obj) {
+    for (var property in obj) {
+      if (obj.hasOwnProperty(property)) {
+        return false;
+      }
+    }
+    return true;
+  };
+
+  var _pushChangeToList = function(valueArray, changeObject) {
```

Comment:
It's not clear when this function will do things, as the use of 'label' introduces a level of indirection. I suggest you get rid of it and just have an if statement within your main function based on whatever actual property of the change you are inspecting.